java/runtime/src/main/java/io/ray/runtime/gcs/GcsClient.java

```diff
@@ -52,6 +56,51 @@ public GcsClient(String redisAddress, String redisPassword) {
     globalStateAccessor = GlobalStateAccessor.getInstance(redisAddress, redisPassword);
   }
 
+  /**
+   * Get Placement Group by {@link PlacementGroupId}
+   * @param placementGroupId Id of Placement Group.
+   * @return The Placement Group.
+   */
+  public PlacementGroup getPlacementGroupInfo(PlacementGroupId placementGroupId) {
+    byte[] result = globalStateAccessor.getPlacementGroupInfo(placementGroupId);
+    Preconditions.checkNotNull(result,
+        "Get Placement Group table data from GlobalStateAccessor failed.");
+
+    PlacementGroupTableData placementGroupTableData;
+    try {
+      placementGroupTableData = PlacementGroupTableData.parseFrom(result);
+    } catch (InvalidProtocolBufferException e) {
+      throw new RuntimeException("Received invalid Placement Group table protobuf data from GCS.");
+    }
+
+    return PlacementGroupUtils.genPlacementGroupFromPbData(placementGroupTableData);
+  }
+
+  /**
+   * Get all Placement Groups in this cluster.
+   * @return All Placement Groups.
+   */
+  public List<PlacementGroup> getAllPlacementGroupInfo() {
+    List<byte[]> results = globalStateAccessor.getAllPlacementGroupInfo();
+
+    List<PlacementGroup> placementGroups = new ArrayList<>();
+    for (byte[] result : results) {
+      Preconditions.checkNotNull(result,
+          "Get Placement Group table data from GlobalStateAccessor failed.");
+
+      PlacementGroupTableData placementGroupTableData;
+      try {
+        placementGroupTableData = PlacementGroupTableData.parseFrom(result);
+      } catch (InvalidProtocolBufferException e) {
+        throw new RuntimeException(
+          "Received invalid Placement Group table protobuf data from GCS.");
+      }
+
+      placementGroups.add(PlacementGroupUtils.genPlacementGroupFromPbData(placementGroupTableData));
```

Comment:
How about extracting `Preconditions.checkNotNull`, `InvalidProtocolBufferException` and `PlacementGroupUtils.genPlacementGroupFromPbData` to a method to remove duplicated code?

Reply:
ok, I will extract it to `PlacementGroupUtils `